packages/@lwc/ssr-compiler/src/compile-js/index.ts

```diff
@@ -111,14 +107,29 @@ const visitors: Visitors = {
             return;
         }
 
-        const decorators = (node as DecoratatedMethodDefinition).decorators;
-        const decoratedExpression = decorators?.[0]?.expression as Expression;
+        const { decorators } = node;
+        // The real type is a subset of `Expression`, which doesn't work with the `is` validators
+        const decoratedExpression = decorators?.[0]?.expression;
         if (
             is.callExpression(decoratedExpression) &&
             is.identifier(decoratedExpression.callee) &&
             decoratedExpression.callee.name === 'wire'
         ) {
-            catalogWireAdapters(path, state);
+            // Getters and setters are methods in the AST, but treated as properties by @wire
+            // Note that this means that their implementations are ignored!
+            if (node.kind === 'get' || node.kind === 'set') {
+                const methodAsProp = b.propertyDefinition(
+                    node.key,
+                    b.identifier('undefined'),
+                    node.computed,
+                    node.static
+                );
+                methodAsProp.decorators = decorators;
+                path.replaceWith(methodAsProp);
```

Comment:
Two things:

1. We should clone the `node.key` and `node.decorators` for safety (or use `immer`).
2. This is a bit janky because, in the case of e.g. `getterWithSetter`, it replaces the getter but _not_ the setter. So you end up with:

```js
  getterWithSetter;
  set getterWithSetter(v) {
    throw new Error("set getterWithSetter should not be called");
  }
```

...when we could just have `getterWithSetter;` alone.

Unfortunately solving this could require backtracing to find the corresponding setter/getter (since the `@wire` may be declared on the second one). Maybe we could use [an `enter`/`leave` function](https://estree-toolkit.netlify.app/traversal#visitor), or do a pre-traversal `traverse` that merely notes with method declarations are `@wire`-decorated.